src/averageAges.js

```diff
@@ -37,7 +40,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const woman = withChildren
+    ? people.filter(person => (person.sex === 'f'
+    && people.some(child => child.mother === person.name)))
+    : people.filter(person => person.sex === 'f');
+
+  const age = woman.reduce((sum, pers) => (sum + (pers.died - pers.born)), 0);
+
+  return age / woman.length;
```

Comment:
I think u can create function to calculate average age and use it for both function

Reply:
Agree